symengine/dense_matrix.cpp

```diff
@@ -63,6 +63,34 @@ unsigned DenseMatrix::rank() const
     throw NotImplementedError("Not Implemented");
 }
 
+bool DenseMatrix::is_lower() const
+{
+    auto A = *this;
+    unsigned n = A.nrows();
+    for (unsigned i = 1; i < n; ++i) {
+        for (unsigned j = 0; j < i; ++j) {
+            if (neq(*(A.get(i, j)), *zero)) {
```

Comment:
```suggestion
            if (not is_zero(*A.get(i, j))) {
```

Reply:
@isuruf , there's no function is_zero, which accepts Symengine::Basic

Reply:
Have you checked latest master? It should be there in `symengine/number.h`

Reply:
Yes, I saw.
I pulled from symenging/symengine, updated the code, and pushed back.